tests/integration/operator/switch_cr_test.go

```diff
@@ -0,0 +1,287 @@
+		Resource: "istiooperators",
+	}
+	var errs util.Errors
+	conditionF := func() (done bool, err error) {
+		us, err := cs.GetUnstructured(gvr, "istio-system", "test-istiocontrolplane")
+		if err != nil {
+			return false, fmt.Errorf("failed to get istioOperator resource: %v", err)
+		}
+		usIOPStatus := us.UnstructuredContent()["status"].(map[string]interface{})
+		iopStatusString, err := json.Marshal(usIOPStatus)
+		if err != nil {
+			return false, fmt.Errorf("failed to marshal istioOperator status: %v", err)
+		}
+		status := &api.InstallStatus{}
+		jspb := jsonpb.Unmarshaler{AllowUnknownFields: true}
+		if err := jspb.Unmarshal(bytes.NewReader(iopStatusString), status); err != nil {
+			return false, fmt.Errorf("failed to unmarshal istioOperator status: %v", err)
+		}
+		errs = util.Errors{}
+		if status.Status != api.InstallStatus_HEALTHY {
+			errs = util.AppendErr(errs, fmt.Errorf("got IstioOperator status: %v", status.Status))
+			return false, nil
+		}
+
+		for cn, cnstatus := range status.ComponentStatus {
+			if cnstatus.Status != api.InstallStatus_HEALTHY {
+				errs = util.AppendErr(errs, fmt.Errorf("got component: %s status: %v", cn, cnstatus.Status))
+				return false, nil
+			}
+		}
+		return true, nil
+	}
+	if errPoll := wait.Poll(pollInterval, pollTimeOut, conditionF); errPoll != nil {
+		return fmt.Errorf("failed to get IstioOperator status: %v", errPoll)
+	}
+	if errs.ToError() != nil {
+		return fmt.Errorf("IstioOperator status is not healthy: %v", errs.ToError())
+	}
+	return nil
+}
+
+func checkControllerInstallation(t *testing.T, ctx framework.TestContext, istioCtl istioctl.Instance, workDir string, iopFile string) {
+	scopes.CI.Infof(fmt.Sprintf("=== Checking istio installation by operator controller with iop file: %s===\n", iopFile))
+	s, err := image.SettingsFromCommandLine()
+	if err != nil {
+		t.Fatal(err)
+	}
+	iop, err := ioutil.ReadFile(iopFile)
+	if err != nil {
+		t.Fatalf("failed to read iop file: %v", err)
+	}
+	metadataYAML := `
+metadata:
+  name: test-istiocontrolplane
+  namespace: istio-system
+`
+	iopcr, err := util.OverlayYAML(string(iop), metadataYAML)
+	if err != nil {
+		t.Fatalf("failed to overlay iop with metadata: %v", err)
+	}
+	iopCRFile := filepath.Join(workDir, "iop_cr.yaml")
+	if err := ioutil.WriteFile(iopCRFile, []byte(iopcr), os.ModePerm); err != nil {
+		t.Fatalf("failed to write iop cr file: %v", err)
+	}
+	initCmd := []string{
+		"operator", "init",
+		"--wait",
+		"-f", iopCRFile,
+		"--hub=" + s.Hub,
+		"--tag=" + s.Tag,
+	}
+	// install istio using operator controller
+	istioCtl.InvokeOrFail(t, initCmd)
+	cs := ctx.Environment().(*kube.Environment).KubeClusters[0]
+
+	// takes time for reconciliation to be done
+	scopes.CI.Infof("waiting for reconciliation to be done")
+	if err := checkInstallStatus(cs); err != nil {
+		t.Fatalf("IstioOperator status not healthy: %v", err)
+	}
+	if _, err := cs.WaitUntilPodsAreReady(cs.NewPodFetch(IstioNamespace)); err != nil {
```

Comment:
Why would a pod be not ready, but install status says "healthy"? If that happens I would want this to fail actually. The user must have a way to know if the install is complete - this is done by the status. If the status is not correct, we should fail the test. Otherwise we are not testing what the user will do

Reply:
ideally it would not happen, but the test wants to test the controller itself, so if there is something wrong about the statusUpdater code change, then this case would possibly happen and we can catch this issue and fix? But I am open to it.

Reply:
if we get into the case where "Status == healthy, pods != ready" the test will pass today, but in reality there is a bug so it should fail?

Reply:
oh I see your point, so it should not wait until pods are ready, just check whether pods ready are not right after, if it is not ready then it should fail, meaning a possible bug.

Reply:
oh actually there would be some possible small delay between the status healthy and the pods are ready, the reason is that in the reconciler or manifest apply, it reaches healthy status after it process and apply the manifests successfully but it does not know the cluster status. I can either keep a small retry value or create new issue for this specifically?

Reply:
So my whole point is the requirement "As a user, I want to know when my deployment has completed" However the user does that, the test should do the same. This may not exist yet - if so this seems like a P0 feature for operator, as we have had lots of feedback about having no clue what the operator is doing. If it does exist, we must test it.

Based on what you said, it sounds like this does not currently exist, because status is declaring healthy == applied not healthy == ready. If so, it doesn't need to block this PR but lets get an issue created and link from here. Does that make sense?

Reply:
yep agreed, that's what I meant for creating an issue for it specifically, created here to track: https://github.com/istio/istio/issues/22815